### PR TITLE
dcm2mha: remove localizer slices, improve logging

### PR DESCRIPTION
**Pull request by joeranbosma**
- Remove localizer slices (these can be part of the main imaging study)
- Improve logging for dcm2mha: full traceback of errors
- Expose the image orientation from SimpleITK in the metadata (to allow custom mappings with this information)

### CODE REVIEW

File: src/picai_prep/dcm2mha.py [hunk 1]
```diff
@@ -520,9 +524,45 @@ def _read_image(self, path: Optional[PathLike] = None) -> sitk.Image:
         try:
             return self._read_image_sitk(path=path)
         except RuntimeError:
-            return self._read_image_pydicom(path=path)
+            # try again while removing localizer slices
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:527]: please explain why localizer slices might cause a RuntimeError?

File: src/picai_prep/dcm2mha.py [hunk 2]
```diff
@@ -535,7 +575,16 @@ def _update_dicom_list(self, path: Optional[PathLike] = None):
         if self.verify_dicom_filenames:
             self._verify_dicom_filenames()
 
-    def _read_image_sitk(self, path: Optional[PathLike] = None) -> sitk.Image:
+        if filter_localizer_slices:
+            # filter out localizer slices (these are sometimes included with the DIOCM slices of the scan)
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:579]: typo: DICOM

File: src/picai_prep/dcm2mha.py [hunk 3]
```diff
@@ -520,9 +524,45 @@ def _read_image(self, path: Optional[PathLike] = None) -> sitk.Image:
...
-            return self._read_image_pydicom(path=path)
+            # try again while removing localizer slices
+            try:
+                return self._read_image_sitk(path=path, filter_localizer_slices=True)
+            except RuntimeError:
+                # try again with pydicom
+                return self._read_image_pydicom(path=path)
+
+    @staticmethod
+    def _filter_localizer_slices(dicom_slice_paths: List[str]) -> List[str]:
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:535]: I understand the reasoning, but I find it very ugly. Is there a way to neatly combine this with lines 615-619?

**Review by snorthman — CHANGES_REQUESTED** [review]

File: src/picai_prep/dcm2mha.py (hunk 1 above)
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py:527]: When the slices in an folder do not form a 3D volume, then SimpleITK throws an `RuntimeError`. This happens e.g. when one/multiple localizer slices are added to a normal scan's folder, or when the folder contains only localizer slices.

File: src/picai_prep/dcm2mha.py (hunk 2 above)
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py]: Fixed

File: src/picai_prep/dcm2mha.py (hunk 3 above)
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py:535]: Fair enough. Thinking about it a bit more resulted in this way: https://github.com/DIAGNijmegen/picai_prep/pull/40
What do you think?

**Review by snorthman — APPROVED** [review]

**Pull request merged**